### DISCUSSION TITLE
Properties added via pipelines are unable to be retieved via MapInputName attribute

### DISCUSSION THREAD

**Discussion by ronnievisser** [spatie/laravel-data#561] [Q&A]
Hi,

I'm trying to process external API responses with this package but facing some issues using it. I'm not sure if this is an issue or my lack of knowledge.

so we have a DTO

```php
class Event extends Data
{
    public function __construct(
        #[MapInputName('customId')]
        public EventIdentifier $id,
        #[MapInputName('participant')]
        public EventParticipant $participant,
    )
    {
    }
}
```

And that DTO has a pipeline:

```php
/**
     * @return DataPipeline
     */
    public static function pipeline(): DataPipeline
    {
        return DataPipeline::create()
            ->into(static::class)
            ->through(EventOrderDataPipe::class)
            ->through(EventIdentifierDataPipe::class);
    }
```

The code of the EventIdentifierDataPipe class is as follows:
```php
public function handle(mixed $payload, DataClass $class, Collection $properties): Collection
    {
        $properties->put('customId', EventIdentifier::from(['id' => (int) $properties->get('id')]));

        return $properties;
    }
```

now what I would have expected is that 'customId' is added to the properties and can be retrieved via the MapInputName attribute. unfortunately it is not doing that but instrad is just taking the id from the response which is also available and is a string.

Is my perception of the functionality wrong? of is this an issue within the package?

Thanks

**Comment by ronnievisser** [comment]
It seems like that the property name itself takes higher presedence then the MapInputName. It that how the package was designed?

**Comment by rubenvanassche** [comment]
You've created a new pipeline but didn't add the default pipes which do the mapping. Take a look at the pipes within the base data class and add these that should probably fix the problem.
